### DISCUSSION TITLE
Do I need to add an id to my patterns if my EntityRuler only has one label?

### DISCUSSION THREAD

**Discussion by Vfgandara** [explosion/spaCy#9424] [Help: Coding & Implementations] [answered]
Hi there,
I'm going to use an EntityRuler to add a pipe to my pipeline. This EntityRuler will basically try to match a lot of patterns (~1500) and assign to the matched entities a single label. Looking at the documentation at [the EntityRuler's page](https://spacy.io/usage/rule-based-matching#entityruler-ent-ids) I saw that one parameter that could be added to each pattern is an `id`. From the documentation alone I couldn't understand exactly what's the purpose of this ID, especially in this case where all my patterns will assign the same label to the Spans. Furthermore, I'll use this pipeline to generate a big `DocBin()` object that will serve as the training data of a NER model that will also only try to predict one single label. In this case, is there a reason for me to add this ID? What will change if I do so or if I don't?

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @Vfgandara, 

> From the documentation alone I couldn't understand exactly what's the purpose of this ID... in this case where all my patterns will assign the same label to the Spans.

IDs are great if you want to link entities or disambiguate them further. Just like in the San Francisco example, most people will call it "san fran," "san francisco," or "sf", but they're all just the same entity--San Francisco. 

If you think that this information (i.e., how entities are linked / disambiguated)  won't be needed for any downstream tasks, then it's fine to leave the `id` parameter empty.
